src/pymor/algorithms/lrradi.py

```diff
@@ -0,0 +1,327 @@
+# This file is part of the pyMOR project (http://www.pymor.org).
+# Copyright 2013-2019 pyMOR developers and contributors. All rights reserved.
+# License: BSD 2-Clause License (http://opensource.org/licenses/BSD-2-Clause)
+
+import scipy.linalg as spla
+import numpy as np
+
+from pymor.algorithms.genericsolvers import _parse_options
+from pymor.algorithms.riccati import _solve_ricc_check_args
+from pymor.vectorarrays.constructions import cat_arrays
+from pymor.core.defaults import defaults
+from pymor.core.logger import getLogger
+from pymor.operators.constructions import IdentityOperator
+from pymor.algorithms.gram_schmidt import gram_schmidt
+from pymor.tools.random import get_random_state
+
+
+@defaults('lrradi_tol', 'lrradi_maxiter', 'lrradi_shifts', 'hamiltonian_shifts_init_maxiter',
+          'hamiltonian_shifts_init_seed')
+def ricc_lrcf_solver_options(lrradi_tol=1e-10,
+                             lrradi_maxiter=500,
+                             lrradi_shifts='hamiltonian_shifts',
+                             hamiltonian_shifts_init_maxiter=20,
+                             hamiltonian_shifts_init_seed=None,
+                             hamiltonian_shifts_subspace_columns=6):
+    """Returns available Riccati equation solvers with default solver options.
+
+    Parameters
+    ----------
+    lrradi_tol
+        See :func:`solve_ricc_lrcf`.
+    lrradi_maxiter
+        See :func:`solve_ricc_lrcf`.
+    lrradi_shifts
+        See :func:`solve_ricc_lrcf`.
+    hamiltonian_shifts_init_maxiter
+        See :func:`hamiltonian_shifts_init`.
+    hamiltonian_shifts_init_seed
+        See :func:`hamiltonian_shifts_init`.
+    hamiltonian_shifts_subspace_columns
+        See :func:`hamiltonian_shifts`.
+    Returns
+    -------
+    A dict of available solvers with default solver options.
+    """
+    return {'lrradi': {'type': 'lrradi',
+                       'tol': lrradi_tol,
+                       'maxiter': lrradi_maxiter,
+                       'shifts': lrradi_shifts,
+                       'shift_options':
+                       {'hamiltonian_shifts': {'type': 'hamiltonian_shifts',
+                                               'init_maxiter': hamiltonian_shifts_init_maxiter,
+                                               'init_seed': hamiltonian_shifts_init_seed,
+                                               'subspace_columns': hamiltonian_shifts_subspace_columns}}}}
+
+
+def solve_ricc_lrcf(A, E, B, C, trans=False, options=None):

```

Comment:
I feel like `R=None` and `S=None` should be here, to have a similar signature as `solve_ricc_lrcf` in `algorithms.riccati` and `bindings.pymess`.